devel-tools/vignetting/sample_focal_plane.py

```diff
@@ -0,0 +1,55 @@
+"""
+Script to create instcat star entries that sample the LSSTCam focal plane.
+"""
+from collections import defaultdict
+import numpy as np
+import pandas as pd
+import galsim
+from lsst.afw import cameraGeom
+import lsst.geom
+import imsim
+
+# Pointing info for DC2 visit 182850:
+ra = 51.99085849573259566
+dec = -40.31737846575015283
+rottelpos = 28.8262515
+mjd = 59822.28563761110854102
+band = 'i'
```

Comment:
Does the vignetting change much for different bands?  Are we ok with just using i band here?

Reply:
I ran the focal plane sampling simulation for `u, g, i, y` bands.  There is up to a ~few percent difference in vignetting response comparing u-band to y-band around 330 mm from the focal plane center:
![vignetting_band_dependence](https://user-images.githubusercontent.com/1994473/204353910-48b8a8b6-6d5d-42b6-bd42-fdb5131f3c35.png)
The top plot shows the spline fits for the `ugiy` sims with the measured fluxes for g-band plotted.  It's zoomed in on the region of the greatest disparity between spline fits. The bottom plot shows the ratio of the spline fits in `ugi` to `y` band.  Note that each spline function is normalized to unity at the center of the focal plane.

It would be straight-forward to have vignetting functions for each band, if we all agree that is worth doing.

Reply:
I think for now, we should change this line to get the band from the instance catalog (filter=3), as that is how the README says to change the filter.  We should make sure `band` here always matches.

Reply:
In that case, we should just get all of the observation info from the main instance catalog file.

Reply:
Sure.  Do we even need to use the instance catalog for this?  We could just specify everything in the config file directly.

Reply:
I actually did try that, but I couldn't get it to work properly--the objects didn't sample the CCDs uniformly.  If you can provide a config file that does that, I'm happy to use that instead.

Reply:
Huh.  That's more variation than I expected, though I admit I've never actually tried to check before.  The filters are slightly powered optics, so it's certainly possible for there to be some small effect on the vignetting curve.